addons/popochiu/editor/importers/aseprite/docks/animation_player_inspector_dock.gd

```diff
@@ -0,0 +1,389 @@
+@tool
+extends PanelContainer
+
+const RESULT_CODE = preload("res://addons/popochiu/editor/config/result_codes.gd")
+const LOCAL_OBJ_CONFIG = preload("res://addons/popochiu/editor/config/local_obj_config.gd")
+# TODO: import a class without breaking coding standards... how?
+const AnimationTagRow = preload("res://addons/popochiu/editor/importers/aseprite/docks/animation_tag_row.gd")
+
+
+var scene: Node
+var target_node: Node
+var config: RefCounted
+var file_system: EditorFileSystem
+
+
+# External logic
+var _animation_creator = preload("res://addons/popochiu/editor/importers/aseprite/animation_creator.gd").new()
+var _animation_tag_row_scene: PackedScene = preload('res://addons/popochiu/editor/importers/aseprite/docks/animation_tag_row.tscn')
+
+# Importer parameters variables
+var _source: String = ""
+var _tags_cache: Array = []
+var _animation_player_path: String
+var _file_dialog_aseprite: FileDialog
+var _output_folder_dialog: FileDialog
+var _importing := false
+var _output_folder := ""
+var _out_folder_default := "[Same as scene]"
+
+# Interface elements
+
+# UI Containers
+@onready var _importer_container = $margin/Importer
+@onready var _warning_container = $margin/Warning
+
+# Title bars, to address theme color problems
+@onready var _options_title_bar = $margin/Importer/OptionsTitleBar
+@onready var _options_title = $margin/Importer/OptionsTitleBar/OptionsTitle
+@onready var _tags_title_bar = $margin/Importer/TagsTitleBar
+@onready var _tags_title = $margin/Importer/TagsTitleBar/TagsTitle
+
+# Source section
+@onready var _source_field = $margin/Importer/Source/SourceButton
+@onready var _rescan_source_button = $margin/Importer/Source/RescanButton
+
+# Tags section
+@onready var _tags_ui_container = $margin/Importer/Tags
+
+# Options section
+@onready var _options_container = $margin/Importer/Options
+@onready var _out_folder_field = $margin/Importer/Options/OutFolder/OutFolderButton
+@onready var _out_filename_field = $margin/Importer/Options/OutFile/OutFileName
+@onready var _visible_layers_field = $margin/Importer/Options/VisibleLayers/VisibleLayersCheckButton
+@onready var _wipe_old_animations_field = $margin/Importer/Options/WipeOldAnimations/WipeOldAnimationsCheckButton
+
+# Warning section
+@onready var _warning_panel = $margin/Warning/HBoxContainer/Panel
+@onready var _warning_panel_label = $margin/Warning/HBoxContainer/Panel/Label
```

Comment:
A small improvement here that could help in case the scene' structure changes in future improvements could be to use [unique nodes](https://docs.godotengine.org/en/stable/tutorials/scripting/scene_unique_nodes.html) to access child nodes.

Reply:
OK, doing it right away!

Reply:
Done, no more wild variables around.